src/main/java/com/solvd/micro9/scooters/service/impl/KStreamsServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package com.solvd.micro9.scooters.service.impl;
+
+import com.solvd.micro9.scooters.messaging.KStreamConfig;
+import com.solvd.micro9.scooters.service.KStreamsService;
+import lombok.RequiredArgsConstructor;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.StoreQueryParameters;
+import org.apache.kafka.streams.state.QueryableStoreTypes;
+import org.apache.kafka.streams.state.ReadOnlyKeyValueStore;
+import org.springframework.kafka.config.StreamsBuilderFactoryBean;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+
+@Service
+@RequiredArgsConstructor
+public class KStreamsServiceImpl implements KStreamsService {
```

Comment:
@eyakauleva 
naming implementations like this you are killing the logical possibility to wrap the objects into chain of objects, a.k.a decorator pattern:

```java
interface Users {

void save(String name);
}

class Default implements Users {
  private final Database db;
  @Override
  void save(String user) {
    this.db.save(user);
  }
}

class Validated implements Users {
  private final Database db;
  private final Users users;
  @Override
  void save(String user) {
  if (!this.db.exists(user)) {
      this.users.save(user);
    }
  }
}

class TxUsers implements Users {
  private final Users users;
  @Transactional
  @Override
  void save(String user) {
    this.users.save(user);
  }
}
```

```java
// spring will construct them into object chain with @Qualifier:
Database db = new Database();
new TxUsers(
  new Validated(
    db,
    new Default(
      db
     )
    ) 
  )
).save(username);
```

thus, all your objects will be small, cohesive and easy to maintain (logic will scale horizontally among objects)